lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -891,6 +891,93 @@ OpFoldResult OrOp::fold(ArrayRef<Attribute> constants) {
       constants, [](APInt &a, const APInt &b) { a |= b; });
 }
 
+/// Merge concats in an or op when they have non-conflicting fields.
+static bool canonicalizeOrExclusiveConcat(OrOp op, PatternRewriter &rewriter) {
+  auto inputs = op.inputs();
+  auto size = inputs.size();
+  auto concat1 = inputs.back().getDefiningOp<ConcatOp>();
+  auto concat2 = inputs[size - 2].getDefiningOp<ConcatOp>();
```

Comment:
How aggressive should this be? only the tail?  all pairs? something in between?  Since you check early for overlap, going over all pairs might be fine.

Reply:
Sounds good, I can check all pairs, and if that ends up being too slow in some cases, we can put some limit on it.

Reply:
My latest commit now iterates over all pairs of `ConcatOp`s.